### PR TITLE
Feature/test

### PR DESCRIPTION
**Pull request by 21ajayns**
Testing for TaskController is completed with Request classes.

### CODE REVIEW

File: app/Http/Controllers/TaskController.php [hunk 1]
```diff
@@ -51,11 +57,10 @@ public function show($id)
      * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function update(Request $request, $id)
+    public function update(TaskCreateRequest $request, $id): Response
```
Comment (boneymylady) [inline app/Http/Controllers/TaskController.php]: Is it `TaskUpdateRequest`?

File: tests/Feature/Http/Controllers/TaskControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        ]);
+        $expected = [
+            'name' => 'test task',
+            'description' => 'this is a test task',
+            'status' => 'on progress'
+        ];
+        $response->assertStatus(201)
+             ->assertJsonFragment($expected);
+        $this->assertDatabaseHas('tasks',$expected);
+
```
Comment (boneymylady) [inline tests/Feature/Http/Controllers/TaskControllerTest.php]: Unwanted extra line

**Review by boneymylady — CHANGES_REQUESTED** [review]
A few minor changes.. Overall, :ok_hand:

File: app/Respositories/Interfaces/TaskRespositoryInterface.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    public function createTask(array $data): Task;
+
+    public function show($id): Task;
+
+    public function update(array $data, $id): Task;
+
+    public function destroy($id): void;
+
+    public function search($name): Task;
+}
```
Comment (boneymylady) [inline app/Respositories/Interfaces/TaskRespositoryInterface.php]: Missing line after closing parenthesis.

**Review by 21ajayns — COMMENTED** [review]
Changes Accepted

File: app/Repositories/Interfaces/ProductRepositoryInterface.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace App\Repositories\Interfaces;
+
+use App\Models\Product;
+use App\Http\Controllers\ProductController;
+use illuminate\support\Collection;
```
Comment (boneymylady) [inline app/Repositories/Interfaces/ProductRepositoryInterface.php]: Capital `I` not `i` for `Illuminate`

File: app/Repositories/Interfaces/TaskRepositoryInterface.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Repositories\Interfaces;
+
+use App\Models\Task;
+use illuminate\support\Collection;
```
Comment (boneymylady) [inline app/Repositories/Interfaces/TaskRepositoryInterface.php]: same here.

**Review by boneymylady — CHANGES_REQUESTED** [review]

**Review by boneymylady — APPROVED** [review]

**Pull request merged**